Initial v13 support

jbowtie opened the pull request:
Adds support for Foundry v13. There may be some minor visual differences.

This has been tested against both Foundry v12 and v13 - we will attempt to maintain v12 compatibility for a few months.

Genii-Locorum commented:
I don't know if this helps or hinder but for the Deprecated item types I've put this in the item.mjs file in Age of Vikings following advice on the Foundry channel.  

  static async createDialog(data={}, createOptions={}, { types, ...options }={}) {
    //Enter the document types you want to remove from the side bar create option - 'base' is removed in the super
    const invalid = ["wound", "family"];
    if (!types) types = this.TYPES.filter(type => !invalid.includes(type));
    else types = types.filter(type => !invalid.includes(type));
    return super.createDialog(data, createOptions, { types, ...options });
  }

jbowtie commented:
@Genii-Locorum I'd appreciate if you can have a quick look at this one before I merge it for release.

Genii-Locorum commented:
I've taken a run through - focus is on what I can see in the changes, but playing with a few other bits.  So not a massive deep dive and these are relatively minor points so cracking job


**Version 12**
In Chat - non-owners can expand the dice roll - from memory only the owner should be able to do this.
You can create "deprecated" item types from the create new item dialog.  Apologies I should have been more explicit that the fix was for V13 - although types is calculating correctly it doesn't seem to be applied in V12 - might need the old code for V12

**Version 13**
Custom Dice rolls don't expand, the default (e.g. /r 1d10) do
Running in Light Mode the combat tab isn't legible.
Tooltip on Character Creation Mode isn't very legible

jbowtie commented:
> Version 12
> In Chat - non-owners can expand the dice roll - from memory only the owner should be able to do this.

May leave this as a known issue

> You can create "deprecated" item types from the create new item dialog. Apologies I should have been more explicit that the fix was for V13 - although types is calculating correctly it doesn't seem to be applied in V12 - might need the old code for V12

Fixed by conditionally registering the old hook

> Version 13
> Custom Dice rolls don't expand, the default (e.g. /r 1d10) do

Fixed

> Running in Light Mode the combat tab isn't legible.

Could not reproduce

> Tooltip on Character Creation Mode isn't very legible

Fixed
